src/pathToVictory/services/officeMatch.service.ts

```diff
@@ -24,31 +27,197 @@ export class OfficeMatchService {
     private prisma: PrismaService,
     private slack: SlackService,
     private aiService: AiService,
-    private votersService: VotersService,
+    private elections: ElectionsService,
   ) {}
 
-  async searchMiscDistricts(
+  async searchDistrictTypes(
     slug: string,
     officeName: string,
     electionLevel: string,
     electionState: string,
+    subAreaName?: string,
+    subAreaValue?: string,
   ): Promise<string[]> {
-    let searchColumns: string[] = []
+    // 1) Pull valid district types from Elections API (state/year aware)
+    let districtTypes: string[] = []
+    try {
+      const campaign = await this.prisma.campaign.findUnique({
+        where: { slug },
+        select: { details: true },
+      })
+      const details = campaign?.details as { electionDate?: string } | undefined
+      const electionYear =
+        details?.electionDate && String(details.electionDate).length >= 4
+          ? Number(String(details.electionDate).slice(0, 4))
+          : undefined
```

Comment:
String manipulation to pull a date out of a formatted date string?  

Seems very odd and flaky.. 

Let's please use `date-fns` when working w/ dates...

https://codesandbox.io/p/devbox/d8y8qs

<img width="582" height="891" alt="Image" src="https://github.com/user-attachments/assets/1ac89b9e-b455-4af0-be81-f900781f84e7" />